docs/proposals/004-scalability-benchmarking.md

```diff
@@ -0,0 +1,108 @@
+Users of Argo CD are interested to know how to scale Argo CD, what configuration tweaks and deployment options they have, and how far they can push resources (in terms of the number of supported applications, Git repositories, managing clusters, etc.).
+
+While the Argo CD documentation [discusses options](https://argo-cd.readthedocs.io/en/stable/operator-manual/high_availability/#scaling-up) to scale up, the actual process is not clear and, as articulated [in this thread](https://github.com/argoproj/argo-cd/issues/9633), oftentimes a point of confusion for users.
+
+By running large-scale benchmarking, we aim at helping the Argo CD community with the following:
+
+* Give confidence to organizations running at significant scale (_needs to be quantified_) that Argo CD can support their use-case, with empirical evidence.
+* Create clear guidelines for scaling Argo CD based on the key scalability factors.
+* Provide recommendations for which topology is best suited for users based on their needs.
+* Determine what work, if any, is needed to improve the scalability of Argo CD.
+
+### Goals
+1. Create a standard set of repeatable benchmarking procedures to objectively measure the limitations of Argo CD.
+    1. May result in a new `argo-cd-benchmarking` repo under `argoproj-labs` so that anyone can easily replicate it and the development of the procedures can happen outside of the lifecycle for Argo CD (unlike [the current `gen-resources` hack](https://github.com/argoproj/argo-cd/tree/master/hack/gen-resources) in the main project).
+    2. Include detailed test scenarios that account for key scalability factors that allow for easy tweaking of the parameters to simplify testing of alternative scenarios.
+2. Determine the baseline for when tweaking is required on the default configuration (resource allocations and replicas).
+    1. One cluster, Applications in In-cluster, default resource allocations.
+3. Quantify how tweaking of existing parameters (replicas, sharding, parallelism, etc) impacts the performance of Argo CD.
+4. Provide a set of metrics and thresholds that will provide a basis for automatically scaling the Argo CD components and alerting for when performance is being impacted by limitations.
+5. All tooling, recommendations, examples, and scenarios will be vendor-agnostic.
+### Non-Goals
+* This proposal does not intend to cover implementation of any auto-scaling based on the metrics and thresholds determined by the scalability benchmarking. A separate proposal will be used for any auto-scaling enhancements.
+* This proposal does not intend to add testing that determines how a change impacts the scalability of Argo CD based on the benchmarks.
+* We do not intend to analyze cost implications of running different topologies and purely focus on scalability requirements from a technology perspective.
+
+## Proposal
+1.  Create new `argo-cd-benchmarking` repo under `argoproj-labs` and add authors of this proposal as maintainers.
```

Comment:
[The proposal](https://github.com/argoproj/argoproj/pull/196) to create the `argoproj-labs/argo-cd-benchmarking` repo has been created and is ready for review!